bands/models.py

```diff
@@ -59,7 +59,10 @@ def email(self):
 
     @property
     def normalized_group_name(self):
-        return ''.join(e for e in self.group.name.strip().lower() if e.isalnum() or e == ' ').replace(' ', '_')
+        # Remove all special characters, then remove all extra whitespace, then replace spaces with underscores
+        return ' '.join(
+            ''.join(e for e in self.group.name.strip().lower() if e.isalnum() or e == ' ').split()
+        ).replace(' ', '_')
```

Comment:
There is probably a better way to do this, but as long as it works & it's tested, then cool with me.